src/Compiler/FSComp.txt

```diff
@@ -1723,4 +1723,5 @@ featureAccessorFunctionShorthand,"underscore dot shorthand for accessor only fun
 3577,tcOverrideUsesMultipleArgumentsInsteadOfTuple,"This override takes a tuple instead of multiple arguments. Try to add an additional layer of parentheses at the method definition (e.g. 'member _.Foo((x, y))'), or remove parentheses at the abstract method declaration (e.g. 'abstract member Foo: 'a * 'b -> 'c')."
 featureUnmanagedConstraintCsharpInterop,"Interop between C#'s and F#'s unmanaged generic constraint (emit additional modreq)"
 3578,chkCopyUpdateSyntaxInAnonRecords,"This expression is an anonymous record, use {{|...|}} instead of {{...}}."
-3579,alwaysUseTypedStringInterpolation,"Interpolated string contains untyped identifiers. Adding typed format specifiers is recommended."
\ No newline at end of file
+3579,alwaysUseTypedStringInterpolation,"Interpolated string contains untyped identifiers. Adding typed format specifiers is recommended."
+3580,tcUnexpectedFunTypeInUnionCaseField,"Unexpected function type in union case field definition"
```

Comment:
This messaging sounds like you cannot use functions as a DU case.
But you can use them, if wrapper in parentheses - this should be made clear.

Reply:
Sure thing! Just tell me what message you'd prefer. The exact wording isn't a big deal to me. I'm mainly focused on getting you a better syntax structure and error code. 😊

Reply:
I think the message proposal by @smoothdeveloper  above is explicit about it, at least it sounds clear to me.

I would go with that one.